### ISSUE TITLE
Provide Auth0 Guide

### ISSUE THREAD

**Issue by Jcambass** [nsarno/knock#172]
There isn't currently an Auth0 Guide for Knock since Auth0 removed its guides on knock and focused on "raw" jwt. It would be really helpful to bring a guide to this repo. I'm currently struggling myself with setting up the knock with Auth0.

**Comment by Amnesthesia** [comment]
same af

**Comment by Carlows** [comment]
I'm in the same position. Have you guys found any pointers?

**Comment by Amnesthesia** [comment]
@carlows Yes! And it's a bit tricky. I had to override some Knock methods in the BaseController to be able to catch the exception thrown, usually from decoding the JWT. First of all you've got to set your audience properly when authenticating against Auth0, and ensure that your API is using the client ID and secret for the API with that defined audience, then double-check what algorithm you're expecting.

I had problems caused by sending a JWT token that was encrypted with RS256 which required a public and private key on the server side, but by instead using the `accessToken` (rather than `idToken`) from Auth0 and passing the `openid` scope and correct audience, helped with getting Knock to decode the token correctly.

It was a pain to debug, but ensure that you're using the correct algorithm — I'm at work right now but can give some code examples later tonight on how I solved it

**Comment by cidylle** [comment]
@Amnesthesia Any time for those code samples? Would be much appreciated. Running into the same issues trying to use Auth0 with Knock.
